test/PowerShellEditorServices.Test/Language/LanguageServiceTests.cs

```diff
@@ -415,4 +415,17 @@ private FindOccurrencesResult FindSymbolsInFile(ScriptRegion scriptRegion)
                     GetScriptFile(scriptRegion));
         }
     }
+
+    internal static class TaskExtensions
+    {
+        public static async Task<T> RunWithTimeout<T>(this Task<T> task, int timeoutMillis = 10000)
+        {
+            if (await Task.WhenAny(task, Task.Delay(timeoutMillis)) == task)
```

Comment:
Might be a little easier to use the tasks' `AsyncWaitHandle`

```csharp
if (!((IAsyncResult)task).AsyncWaitHandle.WaitOne(timeoutMillis))
{
    throw new TimeoutException();
}

return await task;
```

Don't know which route is better, just giving an alternate

Reply:
Ah! That definitely feels more "as designed" than my abuse of tasks